src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/TelemetryClientExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.FeatureManagement.FeatureFilters;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+{
+    /// <summary>
+    /// Provides extension methods for tracking events with TargetingContext.
+    /// </summary>
+    public static class TelemetryClientExtensions
+    {
+        /// <summary>
+        /// Extension method to track an event with <see cref="TargetingContext"/>.
+        /// </summary>
+        public static void TrackEvent(this TelemetryClient telemetryClient, string eventName, TargetingContext targetingContext, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
+        {
+            ValidateTargetingContext(targetingContext);
+
+            if (properties == null)
+            {
+                properties = new Dictionary<string, string>();
+            }
+
+            AddTargetingProperties(properties, targetingContext);
+
+            telemetryClient.TrackEvent(eventName, properties, metrics);
+        }
+
+        /// <summary>
+        /// Extension method to track an <see cref="EventTelemetry"/> with <see cref="TargetingContext"/>.
+        /// </summary>
+        public static void TrackEvent(this TelemetryClient telemetryClient, EventTelemetry telemetry, TargetingContext targetingContext)
+        {
+            ValidateTargetingContext(targetingContext);
+
+            if (telemetry == null)
+            {
+                telemetry = new EventTelemetry();
+            }
+
+            AddTargetingProperties(telemetry.Properties, targetingContext);
+
+            telemetryClient.TrackEvent(telemetry);
+        }
+
+        private static void ValidateTargetingContext(TargetingContext targetingContext)
+        {
+            if (targetingContext == null)
+            {
+                throw new ArgumentNullException(nameof(targetingContext));
+            }
+        }
+
+        private static void AddTargetingProperties(IDictionary<string, string> properties, TargetingContext targetingContext)
+        {
+            properties["TargetingId"] = targetingContext.UserId;
```

Comment:
I push to avoid one-line private methods. There's not really a saving between a one line method call and just inlining the method implementation. But one thing that does happen is reviewing a method results in the need to jump to the private method.

Reply:
Fair point. To me, the one line had justtt too much that might change in the near future to repeat in 4 functions. 

But we ended up removing 2 of the functions (TrackMetric), so I'll remove this function as well for now. I'll still probably pull the string out though.